### ISSUE TITLE
Only allow a single instance

### ISSUE THREAD

**Issue by SchnWalter** [Sixdsn/terra-terminal#80]
Currently you are allowed to run `terra` a 2nd time, but it fails with this exception.

```
Exception: Can't bind Global Keys
```

I propose adding a Process ID file `/var/run/terra.pid`.  This file should  be verified when the application starts and it should prevent Terra from launching a 2nd instance... and instead it should focus the previous window/instance.

Do we want multiple instances of `terra` when developing?  I don't think it would be useful.

**Comment by Sixdsn** [comment]
Hi,

I guess i removed the lock for one instance of terra before implenting the multi window feature.
So now that we can open multiple windows in the current terra instance, it's not relevant to allow multiple terra instances.
And as it fails to start another instance, we should avoid this.

**Comment by SchnWalter** [comment]
We have multiple terminal windows? I assume that you want to have the terminal on multiple screens, right? How do you control their visibility, will the "F12" show and hide all of them at once?

**Comment by Sixdsn** [comment]
Yup, you can add a new window in the menu Terminals -> New Terminal.
The F12 shortcut make them all show/hide together.

**Comment by SchnWalter** [comment]
Looks like normal users can not write to `/var/run/`, alternatively we could  use `/tmp`, but this is kind of ugly.

We should probably deal with this later since it's not a critical issue.

**Comment by Sixdsn** [comment]
Right now it's fixed as it can't bind twice the global hotkey
do you think it's fair enough to keep it this way?

**Comment by SchnWalter** [comment]
Maybe we should maybe keep this ticket for later... we can have proper IPC and just focus the existing instance when you try to open the application a 2nd time... and maybe also have commands to open new tabs in particular folder; for things like an "Open in Terra Terminal" option in file managers.

**Comment by Sixdsn** [comment]
Agree, i close this one as it's done, but open a new one as a spinoff for enhacement

**Comment by Sixdsn** [comment]
'and maybe also have commands to open new tabs in particular folder; for things like an "Open in Terra Terminal" option in file managers.':
Actually there is an option in the preferences menu where you can choose to open a terminal in a specific directory or the same as the previous and it will be saved in the config file.
The feature "Open in Terra Terminal" you mean when we do a right click in the file manager ws could show this option? and then open a new term with this folder?

**Comment by SchnWalter** [comment]
I don't mean the default directory for new terminals... I meant a CLI parameter to be used by file managers with an option on right click, so that you can open it wherever you want, and then focus the newly created window/tab. But this is for when we cleanup the whole thing. I would first like to see the old ConfigManager gone before we start adding new features.

**Comment by Sixdsn** [comment]
Yep i agree, open a new issue for this feature ;)
